experimental/dds/tree2/src/feature-libraries/forestSummarizer.ts

```diff
@@ -47,12 +50,16 @@ export class ForestSummarizer implements Summarizable {
 	 * @returns a snapshot of the forest's tree as a string.
 	 */
 	private getTreeString(stringify: SummaryElementStringifier): string {
```

Comment:
I do hope we fix that we are converting our tree into strings here at some point. Makes the snapshot hard to read, adds escaping on every `"` character and adds needless overhead. As we are changing the format, now might be a good time to do it.

Reply:
What is the alternative? IIRC, don't we need to do this to deal with handles?

Reply:
We do need to deal with handles somewhere, and this code does it incorrectly.

To correctly integrate with fluid GC, we need to ensure handles are in a format fluid can understand. Seralizing them here, even if we do it with the fluid sralizer correctly, isnt good enough, since we then put this string inside more jsonable data, which then gets passed off to the summarizer. It supports handles in that data, but it won't find the ones nested inside this string.

Also this extra stringification is inefficent as it causes tons of string escaping. I'm pretty sure the patten taken by this code was stolen from the schema summarizing logic I wrote It works this way because the schema doesn't change very often, so its trying to efficiently blob the data and reuse the entire schema as on blob across summaries. Schema never contain handles, so this is valid. Since small changes in the forsst are common, this approach is inefficent, and since it can contain handels ints invalid. It also makes the snapshot json files in our tests hard to read.

Reply:
Also since this is not a new issue in this PR, no need to fix it now, but we really should fix this soon.

Reply:
I actually just talked to @connorskees and he'll be working on fixing this soon!